### PR TITLE
28 04 22

### PR DESCRIPTION
**Pull request by Momcil**
My first pull request

### CODE REVIEW

**Comment by Momcil** [comment]
sorry for the writing mistakes in the commits !

**Comment by CJAkkuec** [comment]
I can't start a proper review of your PR for some reason, so I'd like to just leave some general feedback here:

As a sidenote, you might benefit from reconsidering the stucture of your DetailCard.jsx. I'd suggest handling line 48-57 as part of the DetailCard.jsx, and leave the mapping of the AllProjects Array to the page where the cards will be displayed on. At a later point, your data (that is currently hardcoded into your DetailCard.jsx) will be retrieved from your Backend and will likely be imported inside your App.js and distributed from there to your other pages.

In short: DetailCard.jsx is a component file rather than a page. I'd suggest creating a page (for example: projectPage.js) that handles the mapping of the DetailCard.jsx component.

If that's something you'd prefer dealing with at a later point, that's totally fine. I figured I might give you a little heads-up. : ) Let me know if you have any further questions. Looking good otherwise! 👍

File: client/src/pages/DetailCard.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    }
+]
+
+function ShowAllProjects(){
+    
+    return (
+        <>
+            <h1>Projects List</h1>
+            {AllProjects.map((project) => 
+                <ProjectCard >
```
Comment (mpagels) [inline client/src/pages/DetailCard.jsx]: You're creating here a lot of `<ul><li></li></ul>` elements. Because of the `map` Would recommend moving `ProjectCard` over the line 47.

File: client/src/pages/DetailCard.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import styled from "styled-components";
+
+const AllProjects = [
```
Comment (mpagels) [inline client/src/pages/DetailCard.jsx]: The name of the variable should be starting with a `lowerCase` character.

**Review by mpagels — CHANGES_REQUESTED** [review]
Hey Momchil, very good start here. I found one small "big" issue. Pls check.

File: client/src/components/ShowAllProjects.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                        <p>Project ID: {project.ID}</p>
+                        <p>Client: {project.client}</p>
+                        <p>Status: {project.status}</p>
+                        <p>Employee: {project.employee_name} Epmloyee lork-hours: {project.work_hours}</p>
+                        <p>Start date: {project.start_date} End date: {project.end_date}</p>
+                        <p>Sum of work-hours: {project.sum_hours}</p>
+                     </ProjectCard>
+                )}  
+            </Container> 
+        </ >
```
Comment (PReichetanz) [inline client/src/components/ShowAllProjects.jsx]:
```suggestion
        </>    
```

File: client/src/components/ShowAllProjects.jsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+
+
+function ShowAllProjects({allProjects}){
+    
+    return (
+        <>
+            <h1>Projects List</h1>
+            <Container> 
+                { allProjects.map((project) =>
```
Comment (PReichetanz) [inline client/src/components/ShowAllProjects.jsx]:
```suggestion
                {allProjects.map((project) => 
```

File: client/src/components/ShowAllProjects.jsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    return (
+        <>
+            <h1>Projects List</h1>
+            <Container> 
+                { allProjects.map((project) => 
+                    <ProjectCard key={project.ID}>
+                        <p>Project ID: {project.ID}</p>
+                        <p>Client: {project.client}</p>
+                        <p>Status: {project.status}</p>
+                        <p>Employee: {project.employee_name} Epmloyee lork-hours: {project.work_hours}</p>
```
Comment (PReichetanz) [inline client/src/components/ShowAllProjects.jsx]:
```suggestion
                        <p>Employee: {project.employee_name} Employee work-hours: {project.work_hours}</p>
```

File: client/src/components/ShowAllProjects.jsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+                        <p>Sum of work-hours: {project.sum_hours}</p>
+                     </ProjectCard>
+                )}  
+            </Container> 
+        </ >    
+    )
+}    
+
+export default ShowAllProjects
+
```
Comment (PReichetanz) [inline client/src/components/ShowAllProjects.jsx:28]:
```suggestion
```

File: client/src/components/ShowAllProjects.jsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+const ProjectCard = styled.li`
+    text-align: center ;
+    border: 2px solid green;
+    border-radius: 5px;
+    color: #489CB7;
+    list-style-type: none;
```
Comment (PReichetanz) [inline client/src/components/ShowAllProjects.jsx]: If you do this (which is fine), please make sure your list is still accessible; [have a look here](https://developer.mozilla.org/en-US/docs/Web/CSS/list-style-type#accessibility_concerns)

File: client/src/db.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
+const allProjects = [
+    {
+        ID: 1,
```
Comment (PReichetanz) [inline client/src/db.js:3]: I'd suggest to use lowercase "id" here, because otherwise, you probably at some point will have the issue that you use different spellings here and somewhere else in your app; this would consume quite some time to find the bug ;)

File: client/src/db.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+        project_name: "1st Project",
+        client: "1st Client",
+        status: "In progress",
+        employee_name: "Anton",
+        work_hours: 5, 
+        start_date: "10-01_22",
+        end_date: "10-02-22",
+        sum_hours:5,
```
Comment (PReichetanz) [inline client/src/db.js]:
```suggestion
        sum_hours: 5,
```

File: client/src/db.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        ID: 2,
+        project_name: "2nd Project",
+        client: "2nd Client",
+        status: "In progress",
+        employee_name: ["Petter"],
+        work_hours: [2], 
+        start_date: "10-01_22",
+        end_date: "10-02-22",
+        sum_hours:20,
```
Comment (PReichetanz) [inline client/src/db.js]:
```suggestion
        sum_hours: 20,
```

File: client/src/db.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        ID: 3,
+        project_name: "1st Project",
+        client: "3rd Client",
+        status: "Done",
+        employee_name: "Anton",
+        work_hours: 5, 
+        start_date: "10-01_22",
+        end_date: "10-02-22",
+        sum_hours:10,
```
Comment (PReichetanz) [inline client/src/db.js]:
```suggestion
        sum_hours: 10,
```

**Review by PReichetanz — COMMENTED** [review]
Looks good to me, only some minor issues to think about 🚀

**Review by PReichetanz — APPROVED** [review]
Looks good to me!
